src/prop/prop_engine.h

```diff
@@ -223,7 +229,7 @@ class PropEngine
   TheoryEngine* d_theoryEngine;
 
   /** The decision engine we will be using */
-  DecisionEngine* d_decisionEngine;
+  std::unique_ptr<DecisionEngine> d_decisionEngine;
```

Comment:
Maybe move `d_decisionEngine` above `d_theoryEngine` because `d_theoryEngine` depends on `d_decisionEngine`? If we convert `d_theoryEngine` to a unique_ptr, we will probably have to do that anyway to ensure the correct destruction order.

Reply:
`d_theoryEngine` is owned by `SmtEngine` and that shouldn't change, right?

Reply:
and `TheoryEngine` does not depend on `DecisionEngine` anymore with this PR

Reply:
(that's the whole point actually :))

Reply:
You are right. Sorry, I must've gotten confused between `TheoryEngine` and `TheoryProxy`.